examples/poisson_2d_mapping.py

```diff
@@ -759,16 +765,17 @@ def main( *, test_case, ncells, degree, use_spline_mapping, c1_correction, distr
         if use_spline_mapping:
             geometry = Geometry(filename='geo.h5', comm=MPI.COMM_SELF)
             map_discrete = [*geometry.mappings.values()].pop()
-            V = map_discrete.space
+            Vnew = map_discrete.space
```

Comment:
@yguclu, while I am quite certain that it is necessary to rename this space, maybe you want to check for yourself:

In line 640 of this file it reads `M, S = assemble_matrices(V, mapping, kernel)`.
In particular, M is an element of `StencilMatrix(domain=V.vector_space, codomain=V.vector_space)`.

Unless changed, `V` is being overwritten here.

Using ipython, I think that the intended order of usage was meant to be poisson_2d_mapping -> visualize_matrices -> test_mass_matrix.

In lines 21-24 of test_mass_matrix it reads 
```
rhs = assemble_rhs(V, mapping, f1)

M_inv = inverse(M, 'cg', tol=1e-10, maxiter=100, verbose=True
sol = M_inv @ rhs
```

As `V` would have been overwritten, the codomain of `M` and the space of `rhs` didn't coincide as requested by assertions. This is why I renamed `V` to `Vnew` in this file.
I am a little sceptical, as running poisson_2d_mapping.py -t target -n 10 20 -s -c, as suggested by the visualize_matrices docstring, yields a quite bad result.

Reply:
Your reasoning seems correct to me. We should discuss about the "quite bad" results though :sweat

Reply:
Fixed in  4618079